### PR TITLE
InfluxDB: Upgrade Go client, use data source HTTP client

### LINKED ISSUE #26912: Flux datasource leaking TCP connections
**What happened**:
When running a flux query, 2 tcp connection to influxdb are opened and not closed. Over time, this results in grafana not accepting tcp connections with the following error: 
`http: Accept error: accept tcp [::]:3000: accept4: too many open files; retrying in 1s`

I the following GIF, I refresh a single flux query a few times, resulting in 2 extra open connections to influx each time (command:    `netstat -atn | grep ":8086" | wc -l`

Aug-11-2020 09-07-20

**What you expected to happen**:
Grafana not to crash every other day :-) 

**How to reproduce it (as minimally and precisely as possible)**:
Set up a panel with flux query and monitor TCP connections on port 8086.

**Anything else we need to know?**:
This was already mentioned in #26731, and a fix was released in 7.1.3, but the issue persists.
 
**Environment**:
- Grafana version: 7.1.3
- Data source type & version: InfluxDB 1.8.1 with flux.
- OS Grafana is installed on: Ubuntu
- User OS & Browser: Any
- Grafana plugins: /
- Others:

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Upgrade InfluxDB Go client to v2 (have to use Git tag, since they haven't introduced a v2 sub-directory yet, so can't import v2.0.0) and make it use the data source HTTP client. The latter change fixes a TCP connection leakage issue, which was due to creating a new HTTP client for every query, where closed connections would linger for a while in the TIME_WAIT state (this is normal for closed connections). By using the data source HTTP client, we ensure connections are pooled/re-used.

I've tested the fix, and AFAICT no more than two connections to InfluxDB are created at a time.

**Which issue(s) this PR fixes**:
Fixes #26912.

### CODE REVIEW

File: pkg/tsdb/influxdb/flux/executor.go [hunk 1]
```diff
@@ -33,7 +33,6 @@ func executeQuery(ctx context.Context, query QueryModel, runner queryRunner, max
 		dr.Frames = append(dr.Frames, metaFrame)
 		return
 	}
-	defer tables.Close()
```
Comment (papagian) [inline pkg/tsdb/influxdb/flux/executor.go:36]: it doesn't hard to keep this, right?

Comment (aknuds1) [reply pkg/tsdb/influxdb/flux/executor.go:36]: @papagian Per se, but I found out in an issue thread in the Go client repo that it's actually called internally, so clients aren't supposed to call it after all. See this [comment](https://github.com/influxdata/influxdb-client-go/issues/183#issuecomment-674053677) and the [code](https://github.com/influxdata/influxdb-client-go/pull/184/files/7d3656ca553768bc7ee7d4a4b80993f4b6222271#diff-8b3ced92a99034e241e8d5c5818effa7R228-R239) in question.

File: go.sum [hunk 2]
```diff
@@ -235,8 +238,8 @@ github.com/igm/sockjs-go/v3 v3.0.0 h1:4wLoB9WCnQ8RI87cmqUH778ACDFVmRpkKRCWBeuc+W
 github.com/igm/sockjs-go/v3 v3.0.0/go.mod h1:UqchsOjeagIBFHvd+RZpLaVRbCwGilEC08EDHsD1jYE=
 github.com/inconshreveable/log15 v0.0.0-20180818164646-67afb5ed74ec h1:CGkYB1Q7DSsH/ku+to+foV4agt2F2miquaLUgF6L178=
 github.com/inconshreveable/log15 v0.0.0-20180818164646-67afb5ed74ec/go.mod h1:cOaXtrgN4ScfRrD9Bre7U1thNq5RtJ8ZoP4iXVGRj6o=
-github.com/influxdata/influxdb-client-go v1.3.0 h1:R3hLAFtbXCCsPNjlRY8CD66B2ROYWT3miLYLuTIlbbw=
-github.com/influxdata/influxdb-client-go v1.3.0/go.mod h1:S+oZsPivqbcP1S9ur+T+QqXvrYS3NCZeMQtBoH4D1dw=
+github.com/influxdata/influxdb-client-go v0.0.0-20200814135139-d68836272a45 h1:UePjGzItJZeZ4a4375kxZ6pc7myUjuqlpegERMzVYmo=
```
Comment (wbrowne) [inline go.sum:241]: >v0.0.0

paranoia

File: pkg/tsdb/influxdb/flux/executor.go (hunk 1 above)
Comment (papagian) [reply pkg/tsdb/influxdb/flux/executor.go:36]: I see

**Review by papagian — APPROVED** [review]
LGTM

File: go.sum (hunk 2 above)
Comment (wbrowne) [reply go.sum:241]: Make sense but strange to see!

**Review by kylebrandt — APPROVED** [review]

**Pull request merged**